session-replacement/src/main/java/com/amadeus/session/servlet/BaseSessionTracking.java

```diff
@@ -36,7 +39,13 @@ public void configure(SessionConfiguration configuration) {
 
   @Override
   public String newId() {
-    return idProvider.newId();
+    if (appendTimeStamp) {
+      StringBuilder newId = new StringBuilder().append(idProvider.newId());
```

Comment:
This can be slightly optimized if we first get idProvider.newId() and then create StringBuilder with capacity of length of newid + 11 (exclamation mark + timestamp length)